src/Microsoft.AspNetCore.Blazor.Server/DependencyInjection/ServerSideBlazorServiceCollectionExtensions.cs

```diff
@@ -75,101 +60,68 @@ public static IServiceCollection AddServerSideBlazor(this IServiceCollection ser
             {
                 throw new ArgumentNullException(nameof(services));
             }
 
-            return AddServerSideBlazorCore(services, typeof(TStartup), null);
+            return AddServerSideBlazorCore(services, typeof(TStartup));
         }
 
-        /// <summary>
-        /// Adds Server-Side Blazor services to the service collection.
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
-        /// <param name="startupType">A Blazor startup type.</param>
-        /// <param name="configure">A delegate to configure the <see cref="ServerSideBlazorOptions"/>.</param>
-        /// <returns>The <see cref="IServiceCollection"/>.</returns>
-        public static IServiceCollection AddServerSideBlazor(
-            this IServiceCollection services,
-            Type startupType,
-            Action<ServerSideBlazorOptions> configure)
+        private static IServiceCollection AddServerSideBlazorCore(
+            IServiceCollection services,
+            Type startupType)
         {
-            if (services == null)
-            {
-                throw new ArgumentNullException(nameof(services));
-            }
+            AddStandardServerSideBlazorServices(services);
 
-            if (startupType == null)
+            if (startupType != null)
             {
-                throw new ArgumentNullException(nameof(startupType));
-            }
-
-            return AddServerSideBlazorCore(services, startupType, configure);
-        }
+                // Call TStartup's ConfigureServices method immediately
+                var startup = Activator.CreateInstance(startupType);
+                var wrapper = new ConventionBasedStartup(startup);
+                wrapper.ConfigureServices(services);
 
-        /// <summary>
-        /// Adds Server-Side Blazor services to the service collection.
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
-        /// <param name="configure">A delegate to configure the <see cref="ServerSideBlazorOptions"/>.</param>
-        /// <typeparam name="TStartup">A Blazor startup type.</typeparam>
-        /// <returns>The <see cref="IServiceCollection"/>.</returns>
-        public static IServiceCollection AddServerSideBlazor<TStartup>(
-            this IServiceCollection services,
-            Action<ServerSideBlazorOptions> configure)
-        {
-            if (services == null)
-            {
-                throw new ArgumentNullException(nameof(services));
+                // Configure the circuit factory to call a startup action when each
+                // incoming connection is established. The startup action is "call
+                // TStartup's Configure method".
+                services.Configure<DefaultCircuitFactoryOptions>(circuitFactoryOptions =>
+                {
+                    var endpoint = BlazorHub.DefaultPath; // TODO: allow configuring this
```

Comment:
I'm not suggesting that we fix it in this PR, but this is leading us to a design where you specify the 'path' in both `ConfigureServices` and `Configure`. 

One idea that I had about this `CFOptions` would need a map from path -> type, and type->startup. So during `Add...` we map the type->startup, and during `Use...` we map the path->type

Again, I don't expect that we'll solve it as part of this change, but I think we'd like to prevent you from having the specify the path in two places.

Reply:
Makes sense. Yes, I had been thinking of the "path" as being the identifier for the endpoint and thus the thing that defines correspondence between the config and the middleware.

Hopefully you're right that when we come to support multiple endpoints, we can consider possibly friendlier APIs. It might be that we use `TStartup` as the identifier, though technically someone might want to mount the same `TStartup` app paths two different paths (no idea why, but just that it's technically possible).